src/main/java/it/gov/pagopa/pu/fileshare/service/FileStorerService.java

```
@@ -70,9 +73,39 @@ private Path concatenatePaths(String firstPath, String secondPath) {
 
   private void encryptAndSaveFile(MultipartFile file, Path fileLocation)
     throws IOException {
-    try(InputStream is = file.getInputStream();
-      InputStream cipherIs = AESUtils.encrypt(fileEncryptPassword, is)){
+    try (InputStream is = file.getInputStream();
+         InputStream cipherIs = AESUtils.encrypt(fileEncryptPassword, is)) {
       Files.copy(cipherIs, fileLocation, StandardCopyOption.REPLACE_EXISTING);
     }
   }
+
+  public InputStream decryptFile(String filePath, String fileName) {
+    try {
+      // Build the complete file path
+      File file = Paths.get(filePath, fileName).toFile();
+      if (!file.exists() || !file.isFile()) {
+        log.warn("decryptFile - File [{}] not found", file.getAbsolutePath());
+        return null;
```

here you could throw the FileNotFoundException instead of returning null